### DISCUSSION TITLE
.Net on Windows 10, V1.48 and earlier versions

### DISCUSSION THREAD

**Discussion by efconrad** [perpetualKid/FreeTrainSimulator#114] [General]
Windows 10, 64-bit, 21H2 -- always throws the attached dialog when starting:
![image](https://user-images.githubusercontent.com/110104793/184506199-b3f4077a-a476-4633-b7d3-9e219785129e.png)
Clicking on "OK" does nothing.
Legacy .Net and .Net 4.x are installed
![image](https://user-images.githubusercontent.com/110104793/184506256-0afc91d6-7c17-43a7-aa8f-8a20ac27f709.png)
![image](https://user-images.githubusercontent.com/110104793/184506315-f2301186-132f-4c9f-b344-36161d76d297.png)
Any ideas on what could be preventing it from finding the appropriate framework?

**Comment by perpetualKid** [comment]
I'm not quite sure, it's a bit confusing with all the different versions Microsoft did put out there, but looking at the 3rd screenshot, you may need to install the ".NET Desktop runtime" from here (https://dotnet.microsoft.com/en-us/download/dotnet/3.1) instead of the .NET Core Runtime (which is for console applications only, ie. MultiplayerServer.exe should work for you). 

![image](https://user-images.githubusercontent.com/7158903/184508886-08ec36dd-edeb-4a20-8814-e06b0708e727.png)

Alternatively, if you use the SDK version (which contains all the Runtime stuff as well, also for Desktop), you should use the x64 SDK, whereas you only have an x86 version (23bit) of the SDK installed. 
I guess the OR Menu tries to start in x64 mode, so it will not find/use the x86 SDK which you have installed.

As for `Clicking on "OK" does nothing.` there should be a browser window opening to this URL https://dotnet.microsoft.com/en-us/download/dotnet/3.1, but this is all Windows built-in functionality outside of control of OR.

*** 
On another note, after 1.48 this OR version is moved to .NET 6, so if you use the built-in updater to user newer (develop) builds, you will need to install .NET 6 Runtime as well. There should be a similar dialog like in your first screenshot, but getting you to this page on https://dotnet.microsoft.com/en-us/download. From there again go for Runtime version, and select the desktop runtime.

![image](https://user-images.githubusercontent.com/7158903/184509226-ae653788-16a6-4f9e-8d1d-2c65fb38036b.png)

**Comment by efconrad** [comment]
Ah-ha! working now. I may have been going blind and clicking the wrong download, or when I went to Microsoft's site to search, might have gotten presented with an incorrect link... which probably isn't the first time. :)

Not sure why the download from the prompt was failing; that might be a transient Microsoft problem that I just got lucky enough to experience. (I'm a Windows server and Azure AD admin in working life, and little hiccups like that are all in a day's work...)

Thanks for the help and hopefully if anybody else runs into a similar hitch this will be useful.

**Comment by perpetualKid** [comment]
maybe worth to reference/convert that guidance in a wiki entry, thanks for feedback that it worked!
